### ISSUE TITLE
Error: Cannot use 'in' operator to search for 'config' in undefined

### ISSUE THREAD

**Issue by peteruithoven** [micky2be/superlogin-client#29]
I sometimes get the following error when trying to login using the superlogin-client. 

```
Cannot use 'in' operator to search for 'config' in undefined
```
It only sometimes happens, which makes this quite hard to track down. It seems to be a client side code error from the (transpiled & bundled httpInterceptor function:
``` js 
{
                key: "_httpInterceptor",
                value: function() {
                    var a = this
                      , b = function(b) {
                        var c = a.getConfig()
                          , d = a.getSession();
                        return d && d.token ? a.checkRefresh().then(function() {
                            return i(b.url, c.endpoints) && (b.headers.Authorization = "Bearer " + d.token + ":" + d.password),
                            b
                        }) : Promise.resolve(b)
                    }
                      , c = function(b) {
                        var c = a.getConfig();
                        return "config"in b ? (i(b.config.url, c.endpoints) && 401 === b.status && a.authenticated() && (p.warn("Not authorized"),
                        a._onLogout("Session expired")),
                        Promise.reject(b)) : Promise.reject(b)
                    }
                    ;
                    this._http.interceptors.request.eject(this._httpRequestInterceptor),
                    this._http.interceptors.response.eject(this._httpResponseInterceptor),
                    this._httpRequestInterceptor = this._http.interceptors.request.use(b.bind(this)),
                    this._httpResponseInterceptor = this._http.interceptors.response.use(null , c.bind(this))
                }
            }
```
From line:
``` js
return "config"in b ? (i(b.config.url, c.endpoints) && 401 === b.status && a.authenticated() && (p.warn("Not authorized"),
```

I think what happens is that the [http error response interceptor](https://github.com/micky2be/superlogin-client/blob/master/src/index.js#L90-L104) is sometimes called with `undefined`  as `response` argument. Not sure what could cause this. It might be that we're using basicAuth on our server and it gets an invalid response when trying to do a login request.

**Comment by peteruithoven** [comment]
The above PR doesn't fix the case where the response is simply a string, I sometimes get the response `Unauthorized` from the server, this then results in: 

```
Cannot use 'in' operator to search for 'config' in Unauthorized
```

**Comment by peteruithoven** [comment]
I've updated the PR to also handle this.

**Comment by micky2be** [comment]
wow, I never seen any of those edge case yet.
might be the way your serve handle the http request.

**Comment by micky2be** [comment]
Oh yeah, if you use basicAuth that's probably a problem.

**Comment by micky2be** [comment]
Fix published with v0.4.2

**Comment by peteruithoven** [comment]
Thanks for the quick response
